src/shared/multiline-strings.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Remove leading and trailing empty lines from a string.
+ */
+const trimEmptyLines = (str: string): string =>
+  str.replace(/^\s*\n|\n\s*$/g, '');
+
+/**
+ * Remove leading indentation from each line of a string.
+ */
+function dedent(str: string) {
+  // Match the smallest indentation
+  const match = str.match(/^[ \t]*(?=\S)/gm);
+  const indent = match && Math.min(...match.map(el => el.length));
+
+  if (indent) {
+    const regexp = new RegExp(`^ {${indent}}`, 'gm');
+    return str.replace(regexp, '');
+  }
+
+  return str;
+}
+
+/**
+ * Remove leading and trailing empty lines from a string, then remove leading
+ * indentation from each line.
+ *
+ * This is useful to improve readability for multi-line template literals,
+ * where you don't want to
+ *  1) start the first line right after the first backtick.
+ *  2) indent the string inconsistently with the rest of the code.
+ *
+ * This function allows to move from this:
+ *   function foo(arg) {
+ *     if (arg === 3) {
+ *       console.log(`First line
+ *   Second line
+ *   Third line`);
+ *     }
+ *   }
+ *
+ * to this:
+ *   function foo(arg) {
+ *     if (arg === 3) {
+ *       console.log(normalizeMultiLine(`
+ *         First line
+ *         Second line
+ *         Third line
+ *       `));
+ *     }
+ *   }
+ */
+export function normalizeMultiLine(str: string): string {
```

Comment:
I'm not sure if this would be `multiLine` or `multiline`.

Reply:
"Normalize" is a bit ambiguous as it could refer to several things in the context of a string (case normalization? normalization within a line? normalization of line endings?). How about something like `trimAndDedent`?